mentoring/dashboard.py

```diff
@@ -0,0 +1,385 @@
+
+        visual_repr = None
+        if self.visual_rules:
+            try:
+                rules_parsed = json.loads(self.visual_rules)
+            except ValueError:
+                pass  # JSON errors should be shown as part of validation
+            else:
+                visual_repr = DashboardVisualData(blocks, rules_parsed, self.color_for_value)
+
+        return loader.render_template('templates/html/dashboard.html', {
+            'blocks': blocks,
+            'display_name': self.display_name,
+            'visual_repr': visual_repr,
+            'report_url': self.runtime.handler_url(self, "download_report") if include_report_link else None,
+        })
+
+    def student_view(self, context=None):  # pylint: disable=unused-argument
+        """
+        Standard view of this XBlock.
+        """
+        if not self.mentoring_ids:
+            return Fragment(u"<h1>{}</h1><p>{}</p>".format(self.display_name, _("Not configured.")))
+
+        fragment = Fragment(self.generate_content())
+        fragment.add_css_url(self.runtime.local_resource_url(self, self.css_path))
+        return fragment
+
+    @XBlock.handler
+    def download_report(self, request=None, suffix=None):  # pylint: disable=unused-argument
+        if not self.mentoring_ids:
+            return Response("Not generating report - it would be empty.", status=400)
+        report_html = loader.render_template('templates/html/dashboard_report.html', {
+            'title': self.display_name,
+            'body': self.generate_content(include_report_link=False),
+            'css': loader.load_unicode(self.css_path),
+        })
+        return Response(
+            report_html,
+            content_type='text/html',
+            charset='utf8',
+            content_disposition='attachment; filename=report.html',
+        )
+
+    def validate_field_data(self, validation, data):
+        """
+        Validate this block's field data.
+        """
+        super(DashboardBlock, self).validate_field_data(validation, data)
+
+        def add_error(msg):
+            validation.add(ValidationMessage(ValidationMessage.ERROR, msg))
+
+        try:
+            list(self.get_mentoring_blocks(data.mentoring_ids, ignore_errors=False))
+        except InvalidUrlName as e:
+            add_error(_(u'Invalid block url_name given: "{bad_url_name}"').format(bad_url_name=unicode(e)))
+
+        if data.color_rules:
+            try:
+                self.parse_color_rules_str(data.color_rules, ignore_errors=False)
+            except ValueError as e:
+                add_error(unicode(e))
+
+        if data.visual_rules:
+            try:
+                rules = json.loads(data.visual_rules)
+            except ValueError as e:
+                add_error(_(u"Visual rules contains an error: {error}").format(error=e))
+            else:
+                if not isinstance(rules, dict):
+                    add_error(_(u"Visual rules should be a JSON dictionary/object: {...}"))
```

Comment:
There's not much space in the studio error popup to display a big error message, and rules is likely a large string, so though I'd like to, I don't think it would work from a UI/UX perspective.